src/client/sandbox/event/focus-blur.js

```diff
@@ -333,12 +350,30 @@ export default class FocusBlurSandbox extends SandboxBase {
         if (activeElement !== el)
             withoutHandlers = true;
 
+        // NOTE: we should prevent the native focus event, which appears immediately
+        // after el.blur() call and raise this event via event simulator.
+        var onBodyFocus = (e, dispatched, preventEvent, cancelHandlers, stopPropagation) => {
+            preventEvent();
+            cancelHandlers();
+            stopPropagation();
+        };
+
         if (!withoutHandlers) {
             this.elementEditingWatcher.processElementChanging(el);
             this.elementEditingWatcher.stopWatching(el);
+
+            if (isAsync && activeElement !== curDocument.body)
+                this.listeners.addInternalEventListener(curDocument.body, ['focus'], onBodyFocus);
         }
 
         this._raiseEvent(el, 'blur', () => {
+            this.timersSandbox.deferFunction(() => {
+                if (!withoutHandlers && isAsync && activeElement !== curDocument.body && el !== curDocument.body) {
```

Comment:
it looks that the last condition (`el !== curDocument.body`) should not be here.
We call `addInternalEventListener` with `if(!withoutHandlers && isAsync && activeElement !== curDocument.body)`, so we should remove the listener with the same condition